packages/rule-components/src/RuleTable/RuleTable.js

```diff
@@ -0,0 +1,249 @@
+                cells={columns.map(({ transforms, ...column }) => ({
+                    ...column,
+                    transforms: [
+                        ...transforms || [],
+                        ...!isLoading && sortBy && !column.disableSort ? [ sortable ] : []
+                    ]
+                }))}
+                rows={
+                    isLoading ?
+                        [ ...new Array(5) ].map(() => ({
+                            cells: [{
+                                title: <Skeleton size="lg" />,
+                                props: { colSpan: columns.length + Boolean(actions) }
+                            }]
+                        })) :
+                        flatten(createRows(data, columns, expanded, detail)).filter(Boolean)
+                }
+                {...!isLoading && detail && { onCollapse: this.onCollapse } }
+            >
+                <TableHeader />
+                <TableBody />
+            </Table>
+            <TableToolbar isFooter>
+                <Pagination
+                    {...pagination}
+                    widgetId="pagination-options-menu-bottom"
+                    variant={PaginationVariant.bottom}
+                    dropDirection="up"
+                />
+            </TableToolbar>
+        </section>;
+    }
+}
+
+RuleTable.propTypes = {
+    ariaLabel: PropTypes.string,
+    rules: PropTypes.shape({
+        meta: PropTypes.shape({
+            count: PropTypes.number,
+            offset: PropTypes.number,
+            limit: PropTypes.offset
+        }),
+        data: PropTypes.arrayOf(PropTypes.shape({
+            [PropTypes.string]: PropTypes.any
+        }))
+    }),
+    isLoading: PropTypes.bool,
+    columns: PropTypes.arrayOf(PropTypes.shape({
+        title: PropTypes.node,
+        selector: PropTypes.oneOfType([ PropTypes.string, PropTypes.arrayOf(PropTypes.string), PropTypes.func ]),
+        disableSort: PropTypes.bool
+    })),
+    sort: PropTypes.string,
+    defaultFilters: PropTypes.arrayOf(PropTypes.shape({})),
+    fetchData: PropTypes.func,
+    detail: PropTypes.oneOfType([ PropTypes.func, PropTypes.node ]),
+    actions: PropTypes.arrayOf(PropTypes.shape({
+        title: PropTypes.node,
+        onClick: PropTypes.func
+    })),
+    sortBy: PropTypes.shape({
+        index: PropTypes.oneOfType([ PropTypes.string, PropTypes.number ]),
+        direction: PropTypes.string
+    }),
+    filterValues: PropTypes.shape({
+        [PropTypes.string]: PropTypes.oneOfType([
+            PropTypes.arrayOf(PropTypes.oneOfType([ PropTypes.string, PropTypes.number ])),
+            PropTypes.oneOfType([ PropTypes.string, PropTypes.number ])
+        ])
+    }),
+    toolbarProps: PropTypes.any
+};
+
+RuleTable.defaultProps = {
+    rules: {
+        data: []
+    },
+    columns: [],
+    ariaLabel: 'Rules table',
+    isLoading: false,
+    sortBy: {},
+    filters: {},
+    filterValues: {},
+    fetchData: () => undefined
```

Comment:
What do you think about `noop`, because I see same pattern in various places in code?

Reply:
I prefer this over importing noop because well I'd have to import a constant and use it. I have a macro that writes this automatically. And it is imediatelly visible, you know what the function is doing whereas with noop you'd have to trust that noop is doing the same thing.